monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -169,16 +140,18 @@ def get_exploited():
                 'label': exploited_node['label'],
                 'ip_addresses': exploited_node['ip_addresses'],
                 'domain_name': exploited_node['domain_name'],
-                'exploits': list(set(
-                    [ReportService.EXPLOIT_DISPLAY_DICT[exploit['exploiter']] for exploit in exploited_node['exploits']
-                     if exploit['result']]))
+                'exploits': ReportService.get_exploits_used_on_node(exploited_node)
             }
             for exploited_node in exploited]
 
         logger.info('Exploited nodes generated for reporting')
 
         return exploited
 
+    @staticmethod
+    def get_exploits_used_on_node(node: dict) -> List[str]:
+        return list(set([exploit['exploiter'] for exploit in node['exploits'] if exploit['result']]))
```

Comment:
Why the cast from list to set to list? Do you just want a cleaned up list? Do you really need it to be a list and not Iterable?

Reply:
Not sure why it was made this way, but I'd guess to remove duplicates in the list.